cmd/update-notifier_test.go

```diff
@@ -17,34 +17,46 @@
 package cmd
 
 import (
-	"runtime"
 	"strings"
 	"testing"
 	"time"
-
-	"github.com/fatih/color"
 )
 
 // Tests update notifier string builder.
-func TestUpdateNotifier(t *testing.T) {
+func TestComputeUpdateMessage(t *testing.T) {
 	plainMsg := "You are running an older version of Minio released "
-	colorMsg := plainMsg
-	yellow := color.New(color.FgYellow, color.Bold).SprintfFunc()
-	if runtime.GOOS == "windows" {
-		plainMsg += "3 days from now"
-		colorMsg += yellow("3 days from now")
-	} else {
-		plainMsg += "2 days from now"
-		colorMsg += yellow("2 days from now")
-	}
 
-	updateMsg := colorizeUpdateMessage(minioReleaseURL, time.Duration(72*time.Hour))
+	testCases := []struct {
+		older time.Duration
+		dlURL string
 
-	if !(strings.Contains(updateMsg, plainMsg) || strings.Contains(updateMsg, colorMsg)) {
-		t.Fatal("Duration string not found in colorized update message", updateMsg)
+		expectedSubStr string
+	}{
+		{72 * time.Hour, "my_download_url", "days"},
+		{3 * time.Hour, "https://my_download_url_is_huge/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "hours"},
+		{-72 * time.Hour, "another_update_url", ""},
+		{0, "another_update_url", ""},
```

Comment:
Checking humanize souce code, this is not due to a rounding issue, but it is a deliberate choice in the library.

Introducing a test with 1 second time difference prints a message like below:

```
		┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
		┃ You are running an older version of Minio released now ┃
		┃ Update: my_download_url                                ┃
		┗━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┛
```

In our usage, `now` is printed when the time difference is 1 second (exactly - stable release and current release happening like this has a super low probability). Calling it `now` in the message is not a huge inaccuracy - it is ok. Trying to fix this requires using `humanize.CustomRelTime` with a `magnitudes` argument that is only slightly different from `humanize.defaultMagnitudes` (an unexposed library variable). This is excessively complex for a very minor issue (and I don't really see this as a problem at all).

I suggest ignoring this problem and moving on.